main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    each do |i|
+      yield i
+    end
+    self
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield i, index
+      index + 1
+    end
+  end
+
+  def my_select
+    return unless block_given?
+
+    result = []
+    my_each do |i|
+      result.push(i) if yield i
+    end
+    result
+  end
+
+  def my_all?(*val)
+    result = true
+    if !val[0].nil?
+      my_each { |i| result = false unless val[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false unless yield(i) }
+    end
+    result
+  end
+
+  def my_any?(*val)
+    result = true
+    if !val[0].nil?
+      my_each { |i| result = false unless val[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false unless yield(i) }
+    end
+    result
+  end
+
+  def my_none?(*val)
+    result = false
+    if !val[0].nil?
+      my_each { |i| result = false unless val[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false if yield(i) }
+    end
+    result
+  end
```

Comment:
# my_none?
- [ ] `my_none?` returns true if the block never returns true for all elements

      Failure/Error: expect(array.my_none?(&true_block)).to eq(array.none?(&true_block))

        expected: true
             got: false
##
- [ ] `my_none?` when no block or argument is given returns true only if none of the collection members is true

      Failure/Error: expect(false_array.my_none?).to be true

        expected true
             got false
##
- [ ] `my_none?` when a Regex is passed as an argument returns true if none of the collection matches the Regex

      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)

        expected true
             got false
     
- [ ] `my_none?` when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

        expected true
             got false